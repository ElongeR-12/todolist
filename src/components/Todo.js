import React from 'react'
import { Link } from 'react-router-dom'
const Todo = ({todo, setTodos}) => {
    const fetchTodo = async (id) => {
      const res = await fetch(`http://localhost:5000/todos/${id}`)
      const data = await res.json()
      return data
    }
    const toggleCheckbox = async (id) => {
      const todoToggle = await fetchTodo(id)
      const updTodo = { ...todoToggle, state: !todoToggle.state }
      const res = await fetch(`http://localhost:5000/todos/${id}`, {
        method: 'PUT',
        headers: {
          'Content-type': 'application/json',
        },
        body: JSON.stringify(updTodo),
      })
      const data = await res.json()
      setTodos((state)=>{
        const remainTodo = state.filter((item)=>item.id !== data.id);
        return [...remainTodo, data]
      })
    }
  return (
    <div>
        <p>
          <input
            type="checkbox"
            checked={todo?.state}
            onChange={()=>toggleCheckbox(todo?.id)}
          />
        <span>{todo?.title}</span>
        <Link to={`/todolist/todo/${todo?.id}`}>details</Link>
        </p>
    </div>
  )
}

export default Todo