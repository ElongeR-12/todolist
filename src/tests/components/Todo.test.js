import React from 'react';
import { shallow, mount } from 'enzyme';
import Todo from '../../components/Todo';

describe("Todo test validation",()=>{
  test('state should be true or false', () => {
    const todo = { id:Math.random().toString,
      state:false || true,
      description:"Lorem ipsum dolo",
      title:"title two"}
    const wrapper = shallow(<Todo todo={todo}/>);
    const input = wrapper.find('input');
    expect(input.props().defaultChecked).toEqual(todo.state);
});
test('title should be defined', () => {
    const todo = { id:Math.random().toString,
      state:false,
      description:"Lorem ipsum dolo",
      title:"title two"}
    const wrapper = shallow(<Todo todo={todo}/>);
    const input = wrapper.find('span');
    expect(input.text()).toEqual("title two");
    const todo2 = { id:Math.random().toString,
      state:false,
      description:"Lorem ipsum dolo",
      title:"title uknow"}
      const wrapper2 = shallow(<Todo todo={todo2}/>);
    const input2 = wrapper2.find('span');
      expect(input2.text()).toEqual("title uknow");
      const todo3 = { id:Math.random().toString,
        state:false,
        description:"Lorem ipsum dolo",
        title:""}
        const wrapper3 = shallow(<Todo todo={todo3}/>);
      const input3 = wrapper3.find('span');
        expect(input3.text()).not.toBeNull();
});
})