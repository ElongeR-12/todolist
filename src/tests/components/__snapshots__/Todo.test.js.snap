// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo test validation should test TodoItem component 1`] = `
<div>
  <p>
    <input
      onChange={[Function]}
      type="checkbox"
    />
    <span />
    <Link
      to="/todolist/todo/undefined"
    >
      details
    </Link>
  </p>
</div>
`;
